src/Microsoft.Data.SqlClient/tests/FunctionalTests/SslOverTdsStreamTest.cs

```diff
@@ -0,0 +1,347 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.Tests
+{
+    public static class SslOverTdsStreamTest
+    {
+        [Theory]
+        [InlineData(0),InlineData(3),InlineData(128), InlineData(2048), InlineData(8192)]
+        public static void ReadWrite(int readLimit)
+        {
+            const int EncaspulatedPacketCount = 4;
+            const int PassThroughPacketCount = 5;
+
+            SyncTest(EncaspulatedPacketCount, PassThroughPacketCount, readLimit);
+            SyncCoreTest(EncaspulatedPacketCount, PassThroughPacketCount, readLimit);
+            AsyncTest(EncaspulatedPacketCount, PassThroughPacketCount, readLimit);
+            AsyncCoreTest(EncaspulatedPacketCount, PassThroughPacketCount, readLimit);
+        }
+
+        private static void SyncTest(int encapsulatedPacketCount, int passthroughPacketCount, int maxPacketReadLength)
+        {
+            byte[] input;
```

Comment:
this variable could be simplified by taking them inside  `SetupArrays(encapsulatedPacketCount + passthroughPacketCount, out input, out output);` with a little bit of change as 
`SetupArrays(encapsulatedPacketCount + passthroughPacketCount, out byte[] input, out  byte[]output);`

Reply:
Yes but doing so have little benefit and changes the location of the setup of the test from the decorated entry point that you see in the test explorer to a callee method that you'd have to dig into the test to find. 

For tests I prefer verbosity and clarity over minimalization. Especially in this case where the reason for the test might not be entirely obvious. I really want the test to explain itself if you look at it because the primary reason for looking is that it has failed and in that case the extra information can help you find your mistake.

Reply:
Fair enough. Thanks for the explanation.